### DISCUSSION TITLE
Unable to configure a many-to-many relationship

### DISCUSSION THREAD

**Discussion by satwikkansal** [directus/directus#17031] [Archive]
Hi, I have three tables in my database already,

Categories (primary key: category_id)
Images (primary key: image_id)
CategoryImages (composite primary keys that are also foreign keys; category_id, image_id)

I want to set up collections for the same so that when someone adds a new item to the "Categories" collection, they're also able to provide the images in the same interface. However, I'm not able to figure out how to do that.

Why? because when I open the category_images table as a collection I see no columns, but in reality it has 4 columns (the composite keys, plus created_at and updated_at columns), not sure what's going on 😅

![image](https://user-images.githubusercontent.com/10217535/210943251-49de0e80-e5a3-434d-a9fd-b5f75edd2e12.png)

When I try to set up many-to-many relationships on the categories collection, I get this error, it is weird because the collection does exist

![image](https://user-images.githubusercontent.com/10217535/210943307-5e45e4f6-c445-423c-b6ee-e3675ec2421a.png)

This is how I tried to set it up, which lead to the above error

![image](https://user-images.githubusercontent.com/10217535/210943341-d53a9275-97ae-404d-be5c-f444cbf242e7.png)

When I dismissed that error I got another error, which is also weird, since I'm using the admin user

![image](https://user-images.githubusercontent.com/10217535/210943376-a7a7979b-2081-41c5-8010-d81c7fe9228b.png)

I feel like I'm missing something trivial or doing a silly mistake, please help me correct that :)

**Comment by langy** [comment]
Over 18 months later and I'm experiencing this exact same issue. Directus 11, PostgreSQL 15.

**Reply by rijkvanzanten** [reply]
This is an archived old discussion. If you're running into a bug, please open an issue https://github.com/directus/directus/issues/new?assignees=&labels=&projects=&template=bug_report.yml 🙂
